src/ai/backend/common/service_discovery/service_discovery.py

```diff
@@ -176,7 +176,7 @@ def start(self) -> None:
             self._sweep_unhealthy_services_loop()
         )
 
-    async def close(self) -> None:
+    def close(self) -> None:
```

Comment:
Should this `close()` ensure all tasks are cancelled?

Reply:
Cancel the task, but don't wait for it.